### PR TITLE
UI sprint

### PR DESCRIPTION
**Pull request by trestoh**

### CODE REVIEW

File: Models/Library.cs [hunk 1]
```diff
@@ -245,6 +246,33 @@ void exploreDataFiles()
...
+            if (library.ContainsKey(name))
+                return library[name];
+            else 
+                return null;
+        }
+
+        public void addSampleToLibrary(string name, Measurement sample)
+        {
+            Measurement adjusted = sample;
+            if (sample.wavenumbers[0] == 400 && sample.wavenumbers.Length == 2008 && sample.wavenumbers.Last() == 2407)
```
Comment (mzieg) [inline Models/Library.cs:262]: Not all XS models have the same spectral range.

File: Pages/BluetoothPage.xaml [hunk 2]
```diff
@@ -61,6 +61,7 @@
                       ItemsSource="{Binding bleDeviceList}"
                       IsVisible="{Binding usingBluetooth}"
                       ItemSelected="listView_ItemSelected"
+                      VerticalScrollBarVisibility="Never"
```
Comment (mzieg) [inline Pages/BluetoothPage.xaml:64]: I have had to scroll to find my desired serial number.

File: Pages/SettingsPage.xaml [hunk 3]
```diff
@@ -12,13 +12,93 @@
...
+                                                 HorizontalOptions="EndAndExpand"/>
+                    </StackLayout>
+
+                    <!-- gain -->
+                    <StackLayout Orientation="Horizontal">
+                        <Label Text="Gain (dB)" TextColor="#333" VerticalTextAlignment="Center" MaximumWidthRequest="200"/>
+                        <telerik:RadNumericInput x:Name="gainDb" 
+                                                 Value="{Binding gainDb}" 
+                                                 StringFormat="{}{0:f1}"
+                                                 Minimum="0" Maximum="32" Step="0.1"
```
Comment (mzieg) [inline Pages/SettingsPage.xaml:44]: Analog gain only goes to 30.0

File: Popups/OverlaysPopup.xaml
Comment (mzieg) [inline Popups/OverlaysPopup.xaml:1]: neat

File: ViewModels/ScopeViewModel.cs [hunk 4]
```diff
@@ -54,45 +102,252 @@ public ScopeViewModel()
         if (spec != null && spec.paired)
         {
             libraryLoader = Task.Run(() => library = new Library("library", spec));
+            libraryLoader.Wait();
+            library.LoadFinished += Library_LoadFinished;
+            Task.Run(() => findUserFiles());
```
Comment (mzieg) [inline ViewModels/ScopeViewModel.cs:107]: Nice

File: ViewModels/ScopeViewModel.cs [hunk 5]
```diff
@@ -54,45 +102,252 @@ public ScopeViewModel()
...
+
+        SimpleCSVParser parser = new SimpleCSVParser();
+        Stream s = File.OpenRead(file.AbsolutePath);
+        StreamReader sr = new StreamReader(s);
+        await parser.parseStream(s);
+
+        Measurement m = new Measurement();
+        m.wavenumbers = parser.wavenumbers.ToArray();
+        m.raw = parser.intensities.ToArray();
+        m.excitationNM = 785;
```
Comment (mzieg) [inline ViewModels/ScopeViewModel.cs:303]: This will be a buried bug for other models.

**Review by mzieg — APPROVED** [review]

File: Models/Library.cs (hunk 1 above)
Comment (trestoh) [reply Models/Library.cs:262]: Noted and added to backlog

File: Pages/BluetoothPage.xaml (hunk 2 above)
Comment (trestoh) [reply Pages/BluetoothPage.xaml:64]: You can still scroll, this just gets rid of the bar which is an eyesore on mobile

File: Pages/SettingsPage.xaml (hunk 3 above)
Comment (trestoh) [reply Pages/SettingsPage.xaml:44]: fixed

**Pull request merged**